src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs

```diff
@@ -419,12 +424,17 @@ private void UpdateAlignments()
 
 				if (hasPadding)
 				{
-					_grid.ColumnDefinitions.Insert(gridIndexOffSet, new ColumnDefinition() { Width = new GridLength(padding.Top) });
-					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(padding.Left) });
-					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(1, GridUnitType.Star) });
-					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(padding.Right) });
+					_grid.Padding = new Thickness(padding.Left - spacing <= 0 ? padding.Left : 0, 0, padding.Right - spacing <= 0 ? padding.Right : 0, 0);
 
-					gridIndexOffSet += 1;
+					if (padding.Left - spacing > 0)
+					{
+						_grid.ColumnDefinitions.Insert(gridIndexOffSet, new ColumnDefinition() { Width = new GridLength(padding.Left - spacing) });
+						gridIndexOffSet += 1;
+					}
+
+					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(padding.Top) });
+					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(1, GridUnitType.Star) });
+					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(padding.Bottom) }); 
 				}
```

Comment:
Need a else here to clear padding. (+ same for vertical)

Reply:
The padding is reset each time according to the relation padding / spacing.